projects/plugins/boost/app/lib/class-output-filter.php

```diff
@@ -91,11 +91,18 @@ public function add_callback( $callback ) {
 	 * @return string Buffer data to be flushed to browser.
 	 */
 	public function tick( $buffer, $phase ) {
-		// Don't do anything if we're not support to do any filtering.
+		// Bail early if not filtering.
 		if ( ! $this->is_filtering ) {
 			return $buffer;
 		}
 
+		// Special case: If this is the first, last and only chunk, fake an empty 'previous' chunk to keep processing simple.
+		$is_last_chunk = ( $phase & PHP_OUTPUT_HANDLER_END ) > 0;
+		if ( $is_last_chunk && ! isset( $this->buffered_chunk ) ) {
+			$this->buffered_chunk = '';
+		}
+
```

Comment:
How about instead always setting the `buffered_chunk` to string on class initialization?

Reply:
> What exactly is bitwise doing here `($phase & PHP_OUTPUT_HANDLER_END ) > 0` ?

I added some comments explaining, but that is the callback for PHP's output buffer. The second param ($phase) is a bitmask. I've tried to make the code clearer.

I also simplified the logic in 0953bf9562.

Reply:
I grew up on a diet of C. I had no idea that bitwise `&` needed further explanation.

I've added a comment that explains how `$phase` works and why bitwise `&` is used. Does that cover the understanding gap?

Reply:
💘 Wonderful. I had my suspicion that this is something so obvious to you that you paid no attention to it at all 😄